Source/WebCore/html/HTMLDetailsElement.cpp

```diff
@@ -160,23 +160,53 @@ void HTMLDetailsElement::attributeChanged(const QualifiedName& name, const AtomS
                 root->appendChild(*m_defaultSlot);
                 queueDetailsToggleEventTask(DetailsState::Closed, DetailsState::Open);
                 if (auto& name = attributeWithoutSynchronization(nameAttr); document().settings().detailsNameAttributeEnabled() && !name.isEmpty()) {
-                    Vector<RefPtr<HTMLDetailsElement>> otherElementsInNameGroup;
-                    for (auto& detailsElement : descendantsOfType<HTMLDetailsElement>(rootNode())) {
-                        if (&detailsElement != this && detailsElement.attributeWithoutSynchronization(nameAttr) == name)
-                            otherElementsInNameGroup.append(&detailsElement);
-                    }
-                    for (RefPtr otherDetailsElement : otherElementsInNameGroup)
+                    bool shouldFireMutationEvents = document().shouldFireMutationEvents();
+                    document().setShouldFireMutationEvents(false);
+                    for (RefPtr otherDetailsElement : otherElementsInNameGroup())
                         otherDetailsElement->removeAttribute(openAttr);
+                    document().setShouldFireMutationEvents(shouldFireMutationEvents);
                 }
             } else {
                 root->removeChild(*m_defaultSlot);
                 queueDetailsToggleEventTask(DetailsState::Open, DetailsState::Closed);
             }
         }
-    } else
+    } else {
+        ensureDetailsExclusivityAfterMutation();
         HTMLElement::attributeChanged(name, oldValue, newValue, attributeModificationReason);
+    }
+}
+
+Node::InsertedIntoAncestorResult HTMLDetailsElement::insertedIntoAncestor(InsertionType insertionType, ContainerNode& parentOfInsertedTree)
+{
+    ensureDetailsExclusivityAfterMutation();
+    return HTMLElement::insertedIntoAncestor(insertionType, parentOfInsertedTree);
+}
+
+Vector<RefPtr<HTMLDetailsElement>> HTMLDetailsElement::otherElementsInNameGroup()
+{
+    Vector<RefPtr<HTMLDetailsElement>> otherElementsInNameGroup;
+    for (auto& detailsElement : descendantsOfType<HTMLDetailsElement>(rootNode())) {
+        if (&detailsElement != this && detailsElement.attributeWithoutSynchronization(nameAttr) == attributeWithoutSynchronization(nameAttr))
+            otherElementsInNameGroup.append(&detailsElement);
+    }
+    return otherElementsInNameGroup;
 }
 
+void HTMLDetailsElement::ensureDetailsExclusivityAfterMutation()
+{
+    if (auto& name = attributeWithoutSynchronization(nameAttr); m_isOpen && !name.isEmpty() && document().settings().detailsNameAttributeEnabled()) {
+        bool shouldFireMutationEvents = document().shouldFireMutationEvents();
+        document().setShouldFireMutationEvents(false);
+        for (RefPtr otherDetailsElement : otherElementsInNameGroup()) {
+            if (otherDetailsElement->m_isOpen) {
```

Comment:
It’s not a great pattern that the `details` element caches the value of the `open` attribute with `m_isOpen`. There’s no need to store that in a data member for correct behavior and I doubt that it’s an important performance optimization.

Reply:
> It’s not a great pattern that the `details` element caches the value of the `open` attribute with `m_isOpen`. There’s no need to store that in a data member for correct behavior and I doubt that it’s an important performance optimization.

Thanks — it’s gone now. (And I’m glad to see it go — it had been part of the implementation since before I patched it, and I’d wondered myself whether it could actually be providing any win. And after now looking back at the commit history, I can see it seems to have been added very early on in the implementation — rather than being, say, some optimization that was intentionally later.)